tests/acceptance/user-can-waive-hearings-test.js

```diff
@@ -0,0 +1,172 @@
+  });
+
+  hooks.afterEach(async function() {
+    window.location.hash = '';
+
+    await invalidateSession();
+  });
+
+  test('user can waive hearings on to-review page', async function(assert) {
+    seedMirage(server);
+
+    await authenticateSession();
+
+    await visit('/my-projects/to-review');
+
+    assert.ok('[data-test-button="submitHearing"]');
+
+    assert.notOk(find('[data-test-button="onConfirmOptOutHearing"]'));
+    assert.notOk(find('[data-test-button="closeOptOutHearingPopup"]'));
+
+    await click('[data-test-button="optOutHearingOpenPopup"]');
+
+    assert.ok(find('[data-test-button="onConfirmOptOutHearing"]'));
+    assert.ok(find('[data-test-button="closeOptOutHearingPopup"]'));
+
+    await click('[data-test-button="onConfirmOptOutHearing"]');
+
+    assert.notOk(find('[data-test-button="onConfirmOptOutHearing"]'));
+
+    assert.ok(find('[data-test-hearings-waived-message]'));
+    assert.ok(find('[data-test-button="submitRecommendation"]'));
+
+    await click('[data-test-button="submitRecommendation"]');
+
+    assert.equal(currentURL(), '/my-projects/4/recommendations/add?participantType=CB');
+
+    assert.ok(find('[data-test-hearings-waived-message]'));
+  });
+
+  test('User can waive hearings on the show-project page', async function(assert) {
+    await authenticateSession();
+    // user has to be signed in and assigned to that project (dcp_name matches)
+    const userProject = server.create('project', {
+      id: 1,
+      dcp_name: 'P2012M046',
+    });
+
+    this.server.create('project', {
+      id: 1,
+      dcp_name: 'P2012M046',
+    });
+
+    this.server.create('user', {
+      emailaddress1: 'testuser@planning.nyc.gov',
+      projects: [userProject],
+    });
+
+    seedMirage(server);
+
+    // simulate presence of location hash after OAUTH redirect
+    window.location.hash = '#access_token=test';
+
+    await visit('/login');
+
+    await visit('/projects/5');
+
+    assert.ok(find('[data-test-hearing-rec-shortcuts]'));
+
+    assert.ok(find('[data-test-button-hearing-form]'));
+
+    assert.notOk(find('[data-test-button="onConfirmOptOutHearing"]'));
+    assert.notOk(find('[data-test-button="closeOptOutHearingPopup"]'));
+
+    await click('[data-test-button="optOutHearingOpenPopup"]');
+
+    assert.ok(find('[data-test-button="onConfirmOptOutHearing"]'));
+    assert.ok(find('[data-test-button="closeOptOutHearingPopup"]'));
+
+    await click('[data-test-button="onConfirmOptOutHearing"]');
+
+    assert.notOk(find('[data-test-button-hearing-form]'));
+    assert.ok(find('[data-test-hearings-waived-message]'));
+    assert.ok(find('[data-test-button-to-rec-form]'));
+
+    await click('[data-test-button-to-rec-form]');
+
+    assert.equal(currentURL(), '/my-projects/5/recommendations/add');
+
+    assert.ok(find('[data-test-hearings-waived-message]'));
+  });
+
+  test('user can waive hearings on upcoming page', async function(assert) {
+    seedMirage(server);
```

Comment:
I talked to @godfreyyeung about this too, this can get us into trouble later down the road (it's an antipattern known as spooky action at a distance). Basically we'll want each test to setup its own data context. But that's for later...

Reply:
Okay we can tackle this in another issue